src/analysis/retail/priest/discipline/modules/spells/Indemnity.tsx

```diff
@@ -0,0 +1,110 @@
+import SPELLS from 'common/SPELLS';
+import { TALENTS_PRIEST } from 'common/TALENTS';
+import { SpellLink } from 'interface';
+import Analyzer, { Options, SELECTED_PLAYER } from 'parser/core/Analyzer';
+import Events, { CastEvent } from 'parser/core/Events';
+import BoringSpellValueText from 'parser/ui/BoringSpellValueText';
+import ItemHealingDone from 'parser/ui/ItemHealingDone';
+import Statistic from 'parser/ui/Statistic';
+import STATISTIC_CATEGORY from 'parser/ui/STATISTIC_CATEGORY';
+import STATISTIC_ORDER from 'parser/ui/STATISTIC_ORDER';
+
+import { POWER_WORD_SHIELD_ATONEMENT_DUR } from '../../constants';
+
+import AtonementAnalyzer, { AtonementAnalyzerEvent } from '../core/AtonementAnalyzer';
+
+const INDEMNITY_EXTENSION_DURATION = 3000;
+const EVANG_EXTENSION_DURATION = 6000;
+
+type shieldInfo = {
+  cast: CastEvent;
+  extendedByEvang?: boolean;
+};
+
+class Indemnity extends Analyzer {
+  private atonementHealing: number = 0;
+  shields: shieldInfo[] = [];
```

Comment:
```suggestion
type ShieldInfo = {
  cast: CastEvent;
  extendedByEvang?: boolean;
};

class Indemnity extends Analyzer {
  private atonementHealing: number = 0;
  shields: ShieldInfo[] = [];
```

nitpick: by convention, structure types like this should have uppercase names.